### PR TITLE
Update DLC's python script and a series of Dockerfiles of 3 frameworks

### PR DESCRIPTION
**Pull request by Yajing-Z**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/image.py [hunk 1]
```diff
@@ -57,6 +59,13 @@ def __init__(self, info, dockerfile, repository, tag, to_build, stage, context=N
         self.to_build = to_build
         self.build_status = None
         self.client = APIClient(base_url=constants.DOCKER_URL, timeout=constants.API_CLIENT_TIMEOUT)
+        # registry.get_docker_registry_login(registry_name)
+        # registry_name = "harbor-repo.vmware.com"
+        registry_name = os.getenv("REGISTRY")
+        # print(registry_name)
```
Comment (liuqi) [inline src/image.py]: Remove the debug output

File: src/image.py [hunk 2]
```diff
@@ -57,6 +59,13 @@ def __init__(self, info, dockerfile, repository, tag, to_build, stage, context=N
         self.to_build = to_build
         self.build_status = None
         self.client = APIClient(base_url=constants.DOCKER_URL, timeout=constants.API_CLIENT_TIMEOUT)
+        # registry.get_docker_registry_login(registry_name)
+        # registry_name = "harbor-repo.vmware.com"
```
Comment (liuqi) [inline src/image.py]: Remove?

Comment (liuqi) [inline src/image.py]: Remove?

File: src/image_builder.py [hunk 3]
```diff
@@ -144,8 +144,9 @@ def image_builder(buildspec):
         """
 
         info = {
-            "account_id": str(BUILDSPEC["account_id"]),
-            "region": str(BUILDSPEC["region"]),
+            "account": str(BUILDSPEC["account"]),
+            # "region": str(BUILDSPEC["region"]),
```
Comment (liuqi) [inline src/image_builder.py]: Remove?

Comment (liuqi) [inline src/image_builder.py]: Remove?

File: src/registry.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        docker_config_json_data = json.load(fp)
+        for docker_registry_item in docker_config_json_data["auths"]:
+            if registry_name == docker_registry_item:
+                authorizationData = docker_config_json_data["auths"][docker_registry_item]["auth"]
+                auth_token = b64decode(authorizationData).decode()
+                user_name, password = auth_token.split(":")
+                return user_name, password
+            else:
+                print("No found this registry and unable locate credentials")
+                return
```
Comment (liuqi) [inline src/registry.py:25]: If this is copied from somewhere, be careful about the license, copyright, et al.

Comment (liuqi) [inline src/registry.py:25]: If this is copied from somewhere, be careful about the license, copyright, et al.

File: tensorflow/inference/docker/2.8/py3/Dockerfile.cpu [hunk 5]
```diff
@@ -74,7 +64,6 @@ RUN ${PIP} --no-cache-dir install --upgrade \
     setuptools
 
 RUN ${PIP} install --no-cache-dir \
-    "awscli<2" \
     boto3 \
```
Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/Dockerfile.cpu:67]: Remove?

Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/Dockerfile.cpu:67]: Remove?

File: tensorflow/inference/docker/2.8/py3/Dockerfile.cpu [hunk 6]
```diff
@@ -89,9 +78,16 @@ RUN ${PIP} install --no-cache-dir \
...
 
-RUN curl $TFS_URL -o /usr/bin/tensorflow_model_server \
- && chmod 555 /usr/bin/tensorflow_model_server
+# Install tensorflow_model_server
+RUN echo 'deb [arch=amd64] http://storage.googleapis.com/tensorflow-serving-apt stable tensorflow-model-server tensorflow-model-server-universal' >> /etc/apt/sources.list \
+ && curl https://storage.googleapis.com/tensorflow-serving-apt/tensorflow-serving.release.pub.gpg | apt-key add - \
+ && apt-get update \
+ && apt-get -y install tensorflow-model-server \
+#  && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
```
Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/Dockerfile.cpu:90]: Reliable method to install tensorflow_model_server?

Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/Dockerfile.cpu:90]: Reliable method to install tensorflow_model_server?

File: tensorflow/inference/docker/2.8/py3/Dockerfile.cpu [hunk 7]
```diff
@@ -103,83 +99,7 @@ RUN mkdir -p ${MODEL_BASE_PATH}
 # Create a script that runs the model server so we can use environment variables
 # while also passing in arguments from the docker command line
 RUN echo '#!/bin/bash \n\n' > /usr/bin/tf_serving_entrypoint.sh \
- && echo '/usr/bin/tensorflow_model_server --port=8500 --rest_api_port=8501 --model_name=${MODEL_NAME} --model_base_path=${MODEL_BASE_PATH}/${MODEL_NAME} "$@"' >> /usr/bin/tf_serving_entrypoint.sh \
+ && echo '/usr/bin/tensorflow_model_server --port= --rest_api_port=8501 --model_name=${MODEL_NAME} --model_base_path=${MODEL_BASE_PATH}/${MODEL_NAME} "$@"' >> /usr/bin/tf_serving_entrypoint.sh \
```
Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/Dockerfile.cpu]: why 8500 is removed?

Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/Dockerfile.cpu]: why 8500 is removed?

File: tensorflow/inference/docker/2.8/py3/Dockerfile.cpu [hunk 8]
```diff
@@ -103,83 +99,7 @@ RUN mkdir -p ${MODEL_BASE_PATH}
...
  && chmod +x /usr/bin/tf_serving_entrypoint.sh
 
-COPY deep_learning_container.py /usr/local/bin/deep_learning_container.py
-
-RUN chmod +x /usr/local/bin/deep_learning_container.py
```
Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/Dockerfile.cpu:111]: what does deep_learning_container.py do?

Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/Dockerfile.cpu:111]: what does deep_learning_container.py do?

File: tensorflow/inference/docker/2.8/py3/Dockerfile.cpu [hunk 9]
```diff
@@ -103,83 +99,7 @@ RUN mkdir -p ${MODEL_BASE_PATH}
...
-RUN HOME_DIR=/root \
- && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
- && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
- && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
- && chmod +x /usr/local/bin/testOSSCompliance \
- && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
- && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
- && rm -rf ${HOME_DIR}/oss_compliance*
-
-RUN curl https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-2.8/license.txt -o /license.txt
```
Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/Dockerfile.cpu:122]: we should be careful about the license related files.

Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/Dockerfile.cpu:122]: we should be careful about the license related files.

File: tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu [hunk 10]
```diff
@@ -120,7 +111,6 @@ RUN apt-get update \
  && rm -rf /var/lib/apt/lists/*
 
 RUN ${PIP} install -U --no-cache-dir \
-    "awscli<2" \
     boto3 \
```
Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu:114]: remove?

Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu:114]: remove?

File: tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu [hunk 11]
```diff
@@ -146,80 +139,6 @@ RUN echo '#!/bin/bash \n\n' > /usr/bin/tf_serving_entrypoint.sh \
  && echo '/usr/bin/tensorflow_model_server --port=8500 --rest_api_port=8501 --model_name=${MODEL_NAME} --model_base_path=${MODEL_BASE_PATH}/${MODEL_NAME} "$@"' >> /usr/bin/tf_serving_entrypoint.sh \
  && chmod +x /usr/bin/tf_serving_entrypoint.sh
 
-ADD https://raw.githubusercontent.com/aws/deep-learning-containers/master/src/deep_learning_container.py /usr/local/bin/deep_learning_container.py
-
-RUN chmod +x /usr/local/bin/deep_learning_container.py
```
Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu:151]: ditto

Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu:151]: ditto

File: tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu [hunk 12]
```diff
@@ -146,80 +139,6 @@ RUN echo '#!/bin/bash \n\n' > /usr/bin/tf_serving_entrypoint.sh \
...
- && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
- && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
- && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
- && chmod +x /usr/local/bin/testOSSCompliance \
- && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
- && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
- && rm -rf ${HOME_DIR}/oss_compliance*
-
-RUN curl https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-2.8/license.txt -o /license.txt
-
```
Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu:163]: ditto

Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu:163]: ditto

File: tensorflow/training/docker/2.8/py3/Dockerfile.cpu [hunk 13]
```diff
@@ -149,83 +148,9 @@ RUN ${PIP} install --no-cache-dir -U \
     psutil==5.7.2 \
    "urllib3>1.26.0<1.27.0"
 
-ADD https://raw.githubusercontent.com/aws/deep-learning-containers/master/src/deep_learning_container.py /usr/local/bin/deep_learning_container.py
+FROM common
```
Comment (liuqi) [inline tensorflow/training/docker/2.8/py3/Dockerfile.cpu:150]: ditto

Comment (liuqi) [inline tensorflow/training/docker/2.8/py3/Dockerfile.cpu:150]: ditto

File: tensorflow/training/docker/2.8/py3/cu112/Dockerfile.gpu [hunk 14]
```diff
@@ -272,124 +226,7 @@ RUN ldconfig /usr/local/cuda-11.2/targets/x86_64-linux/lib/stubs \
...
 
-RUN HOME_DIR=/root \
-   && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
-   && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
-   && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
-   && chmod +x /usr/local/bin/testOSSCompliance \
-   && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
-   && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
-   && rm -rf ${HOME_DIR}/oss_compliance*
-
```
Comment (liuqi) [inline tensorflow/training/docker/2.8/py3/cu112/Dockerfile.gpu:283]: ditto

Comment (liuqi) [inline tensorflow/training/docker/2.8/py3/cu112/Dockerfile.gpu:283]: ditto

**Review by liuqi — CHANGES_REQUESTED** [review]
The comments for the changes under src/ and tensorflow/

File: mxnet/buildspec.yml [hunk 15]
```diff
@@ -9,51 +9,51 @@ repository_info:
...
+#       target: config.properties
+#     entrypoint:
+#       source: docker/artifacts/mms-entrypoint.py
+#       target:  mms-entrypoint.py
+#     e3-entrypoint:
+#       source: docker/artifacts/mms-entrypoint.py
+#       target: mms-e3-entrypoint.py
+#     deep_learning_container:
+#       source: ../../src/deep_learning_container.py
+#       target: deep_learning_container.py
```
Comment (liuqi) [inline mxnet/buildspec.yml]: Same questions on entrypoint...

File: mxnet/inference/docker/1.9/py3/Dockerfile.cpu [hunk 16]
```diff
@@ -64,7 +56,7 @@ WORKDIR /
 
 RUN pip install --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org \
  && pip install --no-cache-dir \
-    "awscli<2" \
+    # "awscli<2" \
     boto3 \
```
Comment (liuqi) [inline mxnet/inference/docker/1.9/py3/Dockerfile.cpu:59]: remove?

File: mxnet/inference/docker/1.9/py3/cu112/Dockerfile.gpu [hunk 17]
```diff
@@ -1,20 +1,13 @@
...
-LABEL maintainer="Amazon AI"
+LABEL maintainer="VMware AI"
 LABEL dlc_major_version="1"
 
 ARG PYTHON_VERSION=3.8.10
 ARG MMS_VERSION=1.1.8
-ARG MX_URL=https://aws-mx-pypi.s3.us-west-2.amazonaws.com/1.9.0/aws_mx_cu112-1.9.0b20211227.163-py3-none-manylinux2014_x86_64.whl
+# ARG MX_URL=https://aws-mx-pypi.s3.us-west-2.amazonaws.com/1.9.0/aws_mx_cu112-1.9.0b20211227.163-py3-none-manylinux2014_x86_64.whl
+
+ARG MX_URL=https://opentuna.cn/pypi/web/packages/e4/07/d9a40f4a6347d80cea2ef32c3a2f2a40f150592ca132d2986ce835ecfe7d/mxnet_cu112-1.9.0-py3-none-manylinux2014_x86_64.whl
```
Comment (liuqi) [inline mxnet/inference/docker/1.9/py3/cu112/Dockerfile.gpu:8]: looks like not an reliable source

File: mxnet/training/docker/1.9/py3/cu112/Dockerfile.gpu [hunk 18]
```diff
@@ -23,14 +13,13 @@ ENV PYTHONDONTWRITEBYTECODE=1 \
     LANG=C.UTF-8 \
     LC_ALL=C.UTF-8 \
     DGLBACKEND=mxnet \
-    CUDNN_VERSION=8.1.0.77-1+cuda11.2 \
     NCCL_VERSION=2.8.3-1+cuda11.2 \
-    DEBIAN_FRONTEND=noninteractive
+    DEBIAN_FRONTEND=noninteractive \
+    NVIDIA_REQUIRE_CUDA="cuda>=11.2 brand=tesla,driver>=418,driver<419 brand=tesla,driver>=440,driver<441 driver>=450"
```
Comment (liuqi) [inline mxnet/training/docker/1.9/py3/cu112/Dockerfile.gpu:18]: the requirement is related to specific hardware version?

File: pytorch/buildspec.yml [hunk 19]
```diff
@@ -9,171 +9,112 @@ repository_info:
...
+    tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *CUDA_VERSION, "-", *OS_VERSION, "-example" ]
+    docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /example, /Dockerfile., *DEVICE_TYPE ]
+    # context:
+    #   <<: *TRAINING_CONTEXT
+  BuildPyTorchCPUPTInferencePy3DockerImage:
     <<: *INFERENCE_REPOSITORY
     build: &PYTORCH_CPU_INFERENCE_PY3 false
-    image_size_baseline: 4899
+    enable_test_promotion: false
+    image_size_baseline: 4000
```
Comment (liuqi) [inline pytorch/buildspec.yml]: just to ask, why we need this change?

File: pytorch/inference/docker/1.11/py3/Dockerfile.cpu [hunk 20]
```diff
@@ -105,24 +86,25 @@ ENV PATH="$PATH:/home/.openmpi/bin"
...
 
-
-RUN curl -L -o ~/mambaforge.sh https://github.com/conda-forge/miniforge/releases/download/${MAMBA_VERSION}/Mambaforge-${MAMBA_VERSION}-Linux-x86_64.sh \
- && chmod +x ~/mambaforge.sh \
- && ~/mambaforge.sh -b -p /opt/conda \
- && rm ~/mambaforge.sh \
+RUN curl -L -o ~/miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+ && chmod +x ~/miniconda.sh \
+ && ~/miniconda.sh -b -p /opt/conda \
+ && rm ~/miniconda.sh \
```
Comment (liuqi) [inline pytorch/inference/docker/1.11/py3/Dockerfile.cpu:92]: why replace mambaforge with miniconda?

File: pytorch/inference/docker/1.11/py3/Dockerfile.cpu [hunk 21]
```diff
@@ -135,7 +117,9 @@ RUN rm -rf /opt/conda/lib/libtinfo.so.6
 RUN pip install --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org \
  && ln -s /opt/conda/bin/pip /usr/local/bin/pip3 \
  && pip install --no-cache-dir -U \
-    "awscli<2" \
+    notebook \
+    jupyter \
+    matplotlib \
     boto3 \
```
Comment (liuqi) [inline pytorch/inference/docker/1.11/py3/Dockerfile.cpu:123]: remove?

File: pytorch/inference/docker/1.11/py3/cu113/Dockerfile.gpu [hunk 22]
```diff
@@ -145,25 +133,26 @@ RUN apt-get update \
...
- && ~/mambaforge.sh -b -p /opt/conda \
- && rm ~/mambaforge.sh \
+ RUN curl -L -o ~/miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+ && chmod +x ~/miniconda.sh \
+ && ~/miniconda.sh -b -p /opt/conda \
+ && rm ~/miniconda.sh \
  && /opt/conda/bin/conda update conda \
  && /opt/conda/bin/conda install -c conda-forge \
     python=$PYTHON_VERSION \
+ && /opt/conda/bin/conda install -y \
```
Comment (liuqi) [inline pytorch/inference/docker/1.11/py3/cu113/Dockerfile.gpu:143]: ditto

File: pytorch/training/docker/1.11/py3/Dockerfile.cpu [hunk 23]
```diff
@@ -112,25 +98,26 @@ RUN rm -rf /root/.ssh/ \
...
+RUN curl -L -o ~/miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+ && chmod +x ~/miniconda.sh \
+ && ~/miniconda.sh -b -p /opt/conda \
+ && rm ~/miniconda.sh \
  && /opt/conda/bin/conda update conda \
  && /opt/conda/bin/conda install -c conda-forge \
     python=$PYTHON_VERSION \
+ && /opt/conda/bin/conda install -y \
+    # conda 4.10.0 requires ruamel_yaml to be installed. Currently pinned at latest.
+    ruamel_yaml==0.15.100 \
```
Comment (liuqi) [inline pytorch/training/docker/1.11/py3/Dockerfile.cpu:110]: ditto

File: pytorch/training/docker/1.11/py3/cu113/Dockerfile.gpu [hunk 24]
```diff
@@ -153,27 +116,28 @@ RUN mkdir /tmp/openmpi \
...
 
-RUN ompi_info --parsable --all | grep mpi_built_with_cuda_support:value \
- && curl -L -o ~/mambaforge.sh https://github.com/conda-forge/miniforge/releases/download/${MAMBA_VERSION}/Mambaforge-${MAMBA_VERSION}-Linux-x86_64.sh \
- && chmod +x ~/mambaforge.sh \
- && ~/mambaforge.sh -b -p /opt/conda \
- && rm ~/mambaforge.sh \
+ RUN curl -L -o ~/miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+ && chmod +x ~/miniconda.sh \
+ && ~/miniconda.sh -b -p /opt/conda \
+ && rm ~/miniconda.sh \
```
Comment (liuqi) [inline pytorch/training/docker/1.11/py3/cu113/Dockerfile.gpu:122]: ditto

**Review by liuqi — CHANGES_REQUESTED** [review]
part 2 - for the changes under mxnet/ and pytorch/

File: src/image.py (hunk 1 above)
Comment (Yajing-Z) [reply src/image.py]: Done

File: src/image.py (hunk 2 above)
Comment (Yajing-Z) [reply src/image.py]: Done

Comment (Yajing-Z) [reply src/image.py]: Done

File: src/image_builder.py (hunk 3 above)
Comment (Yajing-Z) [reply src/image_builder.py]: Done

Comment (Yajing-Z) [reply src/image_builder.py]: Done

File: tensorflow/inference/docker/2.8/py3/Dockerfile.cpu [hunk 25]
```diff
@@ -103,83 +99,7 @@ RUN mkdir -p ${MODEL_BASE_PATH}
 # Create a script that runs the model server so we can use environment variables
 # while also passing in arguments from the docker command line
 RUN echo '#!/bin/bash \n\n' > /usr/bin/tf_serving_entrypoint.sh \
- && echo '/usr/bin/tensorflow_model_server --port=8500 --rest_api_port=8501 --model_name=${MODEL_NAME} --model_base_path=${MODEL_BASE_PATH}/${MODEL_NAME} "$@"' >> /usr/bin/tf_serving_entrypoint.sh \
+ && echo '/usr/bin/tensorflow_model_server --port= --port=8500 --rest_api_port=8501 --model_name=${MODEL_NAME} --model_base_path=${MODEL_BASE_PATH}/${MODEL_NAME} "$@"' >> /usr/bin/tf_serving_entrypoint.sh \
```
Comment (liuqi) [inline tensorflow/inference/docker/2.8/py3/Dockerfile.cpu]: --port= --port=8500  ----->  --port=8500

File: .github/ISSUE_TEMPLATE/custom.md [hunk 26]
```diff
@@ -0,0 +1,10 @@
+---
+name: Custom issue template
+about: Describe this issue template's purpose here.
+title: ''
+labels: ''
+assignees: ''
+
+---
+
```
Comment (liuqi) [inline .github/ISSUE_TEMPLATE/custom.md]: Please remove this change.

**Review by liuqi — CHANGES_REQUESTED** [review]
two minor issues

**Comment by Yajing-Z** [comment]
> two minor issues

Fixing these two minor issues! Please review!

**Pull request merged**